<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1433</width>
    <height>1023</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>19</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Лабораторная работа №3</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QListWidget" name="listAlg">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>481</width>
      <height>191</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>19</pointsize>
     </font>
    </property>
    <property name="mouseTracking">
     <bool>false</bool>
    </property>
    <property name="movement">
     <enum>QListView::Static</enum>
    </property>
    <property name="currentRow">
     <number>-1</number>
    </property>
    <item>
     <property name="text">
      <string>ЦДА</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Брезенхем (int)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Брезенхем (float)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Брезенхем (устранение ступенчатости)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Ву</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Библиотечная функция</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="labelAlg">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>0</y>
      <width>301</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>19</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Выберите алгоритм:</string>
    </property>
   </widget>
   <widget class="QLabel" name="x0label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>260</y>
      <width>261</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>19</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Начальная точка:</string>
    </property>
   </widget>
   <widget class="QLabel" name="xlabel">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>240</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>19</pointsize>
     </font>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QLabel" name="ylabel">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>240</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
   </widget>
   <widget class="QLabel" name="x1label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>310</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Конечная точка:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="createLineButton">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>490</y>
      <width>291</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Построить отрезок</string>
    </property>
   </widget>
   <widget class="QPushButton" name="chooseGroundButton">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>360</y>
      <width>341</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Выбрать цвет фона</string>
    </property>
   </widget>
   <widget class="QPushButton" name="chooseLineButton">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>420</y>
      <width>341</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Выбрать цвет отрезка</string>
    </property>
   </widget>
   <widget class="QGraphicsView" name="drawView">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>30</y>
      <width>871</width>
      <height>751</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="cleanButton">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>810</y>
      <width>181</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить</string>
    </property>
   </widget>
   <widget class="QLabel" name="paramLabel">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>570</y>
      <width>421</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Параметры для построения пучка:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="efficientButton">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>810</y>
      <width>381</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Сравнить эфф-сть</string>
    </property>
   </widget>
   <widget class="QGraphicsView" name="backView">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>360</y>
      <width>61</width>
      <height>51</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
   </widget>
   <widget class="QGraphicsView" name="lineView">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>420</y>
      <width>61</width>
      <height>51</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
   </widget>
   <widget class="QLabel" name="coordLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>630</y>
      <width>301</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>19</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Координаты центра:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>610</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>19</pointsize>
     </font>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>610</y>
      <width>21</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
   </widget>
   <widget class="QLabel" name="rLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>690</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Радиус пучка:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="rInput">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>690</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
    <property name="value">
     <number>200</number>
    </property>
   </widget>
   <widget class="QLabel" name="deltaAlphaLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>750</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Угловой шаг:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="deltaAlphaInput">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>750</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="maximum">
     <number>359</number>
    </property>
    <property name="value">
     <number>15</number>
    </property>
   </widget>
   <widget class="QPushButton" name="beamButton">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>810</y>
      <width>281</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Построить пучок</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>560</y>
      <width>521</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="x0Input">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>270</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5000</number>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="x1Input">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>310</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5000</number>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
    <property name="value">
     <number>300</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="y0Input">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>270</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5000</number>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="y1Input">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>310</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5000</number>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
    <property name="value">
     <number>300</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="ykInput">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>640</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5000</number>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
    <property name="value">
     <number>300</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="xkInput">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>640</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-5000</number>
    </property>
    <property name="maximum">
     <number>5000</number>
    </property>
    <property name="value">
     <number>300</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>810</y>
      <width>421</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Исследовать ступенчатость</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1433</width>
     <height>50</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
